src/shop-detail.html

```diff
@@ -271,6 +303,16 @@
         });
       },
 
+      _buyNow: function() {
+        var item  = {
+          item: this.item,
+          quantity: parseInt(this.$.quantitySelect.value, 10),
+          size: this.$.sizeSelect.value
+        };
+
+        this.fire('buy-item', item);
```

Comment:
The `buy-item` event could be handled by `show-app` which then can call `this.$.payment.request();` assuming ```<shop-payment-request id="payment" ...>```

Also `buy-item` might not need the item in the event's detail.